src/labeledinput/labeledinput.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import uid from '../../utils/uid.js';
+
+import Model from '../model.js';
+import Controller from '../controller.js';
+
+import InputText from '../inputtext/inputtext.js';
+import InputTextView from '../inputtext/inputtextview.js';
+
+import InputLabel from '../inputlabel/inputlabel.js';
+import InputLabelView from '../inputlabel/inputlabelview.js';
+
+/**
+ * The labeled input controller class. It contains two components {@link ui.input.InputLabel InputLabel} and
+ * {@link ui.input.InputText InputText} pair each other by {@link ui.input.LabeledInputModel#uid unique id}.
+ *
+ * 		const model = new Model( {
+ *			label: 'Label text'
+ *			value: 'init value',
+ *		} );
+ *
+ *		new LabeledInput( model, new LabeledInputView() );
+ *
+ * See {@link ui.input.labeled.LabeledInputView}.
+ *
+ * @memberOf ui.input.labeled
+ * @extends ui.Controller
+ */
+export default class LabeledInput extends Controller {
+	/**
+	 * Creates an instance of {@link ui.input.labeled.LabeledInput} class.
+	 *
+	 * @param {ui.input.labeled.LabeledInputModel} model Model of this input.
+	 * @param {ui.View} view View of this input.
+	 */
+	constructor( model, view ) {
+		super( model, view );
+
+		// Create unique id to pair input with label.
+		model.set( 'uid', uid() );
```

Comment:
I thought it might be useful to expose this property, because it's possible to attach more than one label to the same input, but input id is still available by `LabeledInput.input.model.id`, so `uid` could be `@protected`.